firestore/vector_search_basic_test.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package firestore
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/firestore"
+	apiv1 "cloud.google.com/go/firestore/apiv1/admin"
+	"cloud.google.com/go/firestore/apiv1/admin/adminpb"
+)
+
+func createVectorIndexes(t *testing.T, collName string,
+	projectID string, dimension int32, fieldPath string) {
+	dbPath := "projects/" + projectID + "/databases/(default)"
+	ctx := context.Background()
+
+	// Create client
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() { client.Close() })
+	collRef := client.Collection(collName)
+
+	// Create admin client
+	adminClient, err := apiv1.NewFirestoreAdminClient(ctx)
+	if err != nil {
+		t.Fatalf("NewFirestoreAdminClient: %v", err)
+	}
+	t.Cleanup(func() { adminClient.Close() })
+
+	indexParent := fmt.Sprintf("%s/collectionGroups/%s", dbPath, collRef.ID)
+
+	// create vector mode indexes
+	req := &adminpb.CreateIndexRequest{
+		Parent: indexParent,
+		Index: &adminpb.Index{
```

Comment:
The larger part of this function seems like it would be fairly common to any test that needs an index, and this index value itself would be the only part that needs to be passed in. Can we have a common function that handles this?

Reply:
Updated